### PR TITLE
Antivirus: Add MediaScan in the attachments handling

### PR DESCRIPTION
**Pull request by yostyle**
#204

### CODE REVIEW

File: vector/src/main/java/im/vector/app/core/extensions/Session.kt [hunk 1]
```diff
@@ -34,9 +34,21 @@ fun Session.configureAndStart(context: Context, startSyncing: Boolean = true) {
         startSyncing(context)
     }
     refreshPushers()
+    configureContentScanner()
     context.singletonEntryPoint().webRtcCallManager().checkForProtocolsSupportIfNeeded()
 }
 
+private fun Session.configureContentScanner() {
+    sessionParams.homeServerConnectionConfig.antiVirusServerUri
+            ?.let { url ->
```
Comment (Florian14) [inline vector/src/main/java/im/vector/app/core/extensions/Session.kt:43]: `?.let` is not useful here (we do not use the resulting value, only check the nullability), so I would prefere to declare a variable and then use a simple if to check the nullability or trigger a warning

File: vector/src/main/java/im/vector/app/core/extensions/Session.kt [hunk 2]
```diff
@@ -34,9 +34,21 @@ fun Session.configureAndStart(context: Context, startSyncing: Boolean = true) {
...
+                contentScannerService().let {
```
Comment (Florian14) [inline vector/src/main/java/im/vector/app/core/extensions/Session.kt:44]: Also I would prefere to use a `run` or `with` as we don't care about the resulting value. Also you can store the service in a dedicated variable

File: vector/src/main/java/im/vector/app/features/home/room/detail/timeline/helper/ContentScannerStateTracker.kt [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+import org.matrix.android.sdk.api.session.contentscanner.ScanStatusInfo
+import org.matrix.android.sdk.api.util.Optional
+import org.matrix.android.sdk.internal.crypto.attachments.ElementToDecrypt
+import timber.log.Timber
+import javax.inject.Inject
+
+class ContentScannerStateTracker @Inject constructor(private val activeSessionHolder: ActiveSessionHolder) {
+
+    private val lifecycleOwner: LifecycleOwner = LifecycleOwner { lifecycleRegistry }
+    private val lifecycleRegistry: LifecycleRegistry = LifecycleRegistry(lifecycleOwner)
```
Comment (Florian14) [inline vector/src/main/java/im/vector/app/features/home/room/detail/timeline/helper/ContentScannerStateTracker.kt:38]: I don't understand how does it work? LifecycleOwner contains a lambda which returns the lifecycleRegistry and this one is created with the lifecycleOwner in its constructor

File: vector/src/main/res/values/strings_content_scanner.xml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+
+    <string name="antivirus_clean">Trusted</string>
+    <string name="antivirus_infected">Infected File</string>
+    <string name="antivirus_in_progress">Scanning…</string>
+
+</resources>
```
Comment (Florian14) [inline vector/src/main/res/values/strings_content_scanner.xml:8]: Please add a new line at the end of the file

File: vector/src/main/res/values-fr/strings_content_scanner.xml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+
+    <string name="antivirus_clean">Vérifié</string>
+    <string name="antivirus_infected">Fichier Bloqué</string>
+    <string name="antivirus_in_progress">Scan en cours…</string>
+
+</resources>
```
Comment (Florian14) [inline vector/src/main/res/values-fr/strings_content_scanner.xml:8]: Please add a new line at the end of the file

File: vector/src/main/res/layout/item_timeline_event_media_message_stub.xml [hunk 6]
```diff
@@ -31,6 +31,40 @@
...
+
+    <TextView
+        android:id="@+id/messageMediaAvText"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:drawablePadding="4dp"
+        android:visibility="gone"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/messageThumbnailView"
```
Comment (Florian14) [inline vector/src/main/res/layout/item_timeline_event_media_message_stub.xml:63]: Pleace remove the `+` we should not be used for referenced ids

File: vector/src/main/res/layout/item_timeline_event_media_message_stub.xml [hunk 7]
```diff
@@ -43,6 +77,6 @@
         android:visibility="gone"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toBottomOf="@+id/messageThumbnailView"
+        app:layout_constraintTop_toBottomOf="@+id/messageMediaAvText"
```
Comment (Florian14) [inline vector/src/main/res/layout/item_timeline_event_media_message_stub.xml:80]: Please remove the + we should not be used for referenced ids

File: vector/src/main/res/layout/item_timeline_event_file_stub.xml [hunk 8]
```diff
@@ -74,7 +87,7 @@
         android:visibility="gone"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toBottomOf="@+id/horizontalBarrier"
+        app:layout_constraintTop_toBottomOf="@+id/messageFileAvText"
```
Comment (Florian14) [inline vector/src/main/res/layout/item_timeline_event_file_stub.xml:90]: Please remove the + we should not be used for referenced ids

File: vector/src/main/res/layout/item_timeline_event_file_stub.xml [hunk 9]
```diff
@@ -64,6 +64,19 @@
...
 
+    <TextView
+        android:id="@+id/messageFileAvText"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:drawablePadding="4dp"
+        android:visibility="gone"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/horizontalBarrier"
```
Comment (Florian14) [inline vector/src/main/res/layout/item_timeline_event_file_stub.xml:75]: Please remove the + we should not be used for referenced ids

File: vector/src/main/res/drawable/ic_av_bug_background.xml [hunk 10]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android">
+    <solid android:color="?colorError" />
+    <corners android:radius="8dp" />
+</shape>
```
Comment (Florian14) [inline vector/src/main/res/drawable/ic_av_bug_background.xml:5]: Please add a new line at the end of the file

File: vector/src/main/java/im/vector/app/features/home/room/detail/timeline/item/ScannableHolder.kt [hunk 11]
```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2020 New Vector Ltd
```
Comment (Florian14) [inline vector/src/main/java/im/vector/app/features/home/room/detail/timeline/item/ScannableHolder.kt:2]: Please change to 2021

File: vector/src/main/java/im/vector/app/core/extensions/Session.kt (hunk 1 above)
Comment (yostyle) [reply vector/src/main/java/im/vector/app/core/extensions/Session.kt:43]: It used at line 46
`it.setScannerUrl(url.toString())`

**Review by giomfo — APPROVED** [review]
Tests ok - LGTM
I just created 2 minor issues #304, #306

**Pull request merged**